portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +33,56 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    ArrayList<String> list = new ArrayList<String>();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String json = convertToJson(list);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+}
+
+@Override
+public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    String comment = request.getParameter("comment");
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("comment", comment);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    /////////////////////////
+
+    Query query = new Query("Comment").addSort("comment");
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String text = (String) entity.getProperty("comment");
+
+      list.add(text);
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(list));
+}
+
+  private String convertToJson(ArrayList<String> data) {
+    String json = "{ \n";
+    json += "\"comment\": ";
+    json += "\"" + data + "\"";
+    json += ", ";
+    json += "\n";  
+    json += "}";
```

Comment:
Rather than using += for each element to append because it is hard to understand what the final string looks like. You might want to use a StringBuilder or simply use String.format(,). String.format might be the cleanest way to achieve what you want.